Gateway/Response/PaymentCaptureDetailsHandler.php

```diff
@@ -38,22 +52,65 @@ public function handle(array $handlingSubject, array $response)
         $payment = $payment->getPayment();
 
-        // set pspReference as lastTransId only!
-        $payment->setLastTransId($response['pspReference']);
-
-        // The capture request will return a capture-received message, but it doesn't mean the capture has been final
-        // so the invoice is set to Pending
-        // The capture request will return a capture-received message, but it doesn't mean the capture has been final
-        // so the invoice is set to Pending
-        if ($response["response"] === '[capture-received]') {
-            $payment->setIsTransactionPending(true);
-
-            /**
-             * close current transaction because you have capture the goods
-             * but do not close the authorisation because you can still cancel/refund order
-             */
-            $payment->setIsTransactionClosed(true);
-            $payment->setShouldCloseParentTransaction(false);
+        if (array_key_exists(TransactionCapture::MULTIPLE_AUTHORIZATIONS, $response)) {
+            $this->handleMultipleCaptureRequests($payment, $response);
+        } else {
+            // set pspReference as lastTransId only!
+            $payment->setLastTransId($response['pspReference']);
+
+            // The capture request will return a capture-received message, but it doesn't mean the capture has been final
+            // so the invoice is set to Pending
+            if ($response["response"] === TransactionCapture::CAPTURE_RECEIVED) {
+                $this->setInvoiceToPending($payment);
+            }
+        }
+    }
+
+    /**
+     * @param $payment
+     * @param $responseContainer
+     */
+    public function handleMultipleCaptureRequests($payment, $responseContainer)
+    {
+        $this->adyenLogger->info(sprintf(
+            'Handling multiple capture response in details handler for payment %s',
+            $payment->getId()
+        ));
+
+        $captureNotReceived = [];
+
+        foreach ($responseContainer[TransactionCapture::MULTIPLE_AUTHORIZATIONS] as $response) {
+            if ($response["response"] !== TransactionCapture::CAPTURE_RECEIVED) {
+                $captureNotReceived[] = $response['pspReference'];
+            }
+            $lastTransId = $response['pspReference'];
+        }
+
+        // Set transId to the last capture request
+        $payment->setLastTransId($lastTransId);
+
+        if (empty($captureNotReceived)) {
+            $this->setInvoiceToPending($payment);
+        } else {
+            $this->adyenLogger->error(sprintf(
+                'Response for transactions [%s] did not contain [capture-received]',
+                implode(', ', $captureNotReceived)
+            ));
         }
     }
+
+    /**
+     * Set payment to pending
+     *
+     * @param $payment
+     * @return mixed
+     */
+    private function setInvoiceToPending($payment)
+    {
+        $payment->setIsTransactionPending(true);
+        $payment->setIsTransactionClosed(true);
```

Comment:
This was working before it was outsourced to this function, but does it make sense? Is it changing the transaction status twice or is it doing something else?

Reply:
2b70e1c I don't think it does. I removed the `setIsTransactionClosed` call and everything still works fine. Also re-added the comment which was previously there